packages/jest-expo/universal/jest-preset.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const reactNativePreset = require('react-native-web/jest-preset');
+const expoPreset = require('../jest-preset');
+
+const COLORS = {
+  android: 'blueBright',
+  ios: 'white',
+  web: 'magenta',
+  node: 'cyan',
+};
+
+function getModuleFileExtensions(...platforms) {
+  let fileExtensions = [];
+
+  // Support both TypeScript and JavaScript
+  for (const language of ['t', 'j']) {
+    // Support JS and JSX
+    for (const extension of [`${language}s`, `${language}sx`]) {
+      // Ensure order is correct: [platformA.js, platformB.js, js]
+      for (const platform of platforms.concat([''])) {
+        fileExtensions.push([platform, extension].filter(Boolean).join('.'));
+      }
+    }
+  }
+  // Always add this last
+  fileExtensions.push('json');
+  return fileExtensions;
+}
+
+function getPlatformPreset(...platforms) {
```

Comment:
It would be better not to conflate the platform name with the extension:
```suggestion
function getPlatformPreset(displayOptions, extensions) {
```

```
getPlatformPreset({ name: 'iOS', color: COLORS.ios }, ['ios', 'native'])
```

Reply:
separate the sub-extensions from the display logic